src/operator.jl

```diff
@@ -13,6 +16,123 @@ for f in (:fragtype_a, :fragtype_b, :fragtype_accum, :load_a, :load_b, :load_c,
+                @unroll for k = 1 : K
+                    y_layout, x_layout = $convert_index_func((y + 4 * (m - 1), x + (k - 1)))
+                    @inbounds frag = setindex(frag, CT(workspace[y_layout, x_layout]), m + (M ÷ 4) * (k - 1))
+                end
+            end
+            
+            return NTuple{M * K ÷ 4, CT}(frag)
+        end
+
+        @inline function load_b(::Type{FPUOp{M, N, K, DT, CT}}, ::Type{$layout_type{CT}}, workspace, tile::Tile) where {M, N, K, DT, CT}
+            laneId = (threadIdx().x - 1) % 32 + 1
+
+            op_x = (laneId - 1) % 8 + 1
+            y, x = (tile.base.K + tile.offset.K + 1, tile.base.N + tile.offset.N + op_x)
+
+            frag = LocalArray{Tuple{K * N ÷ 8}, CT}(undef)
+            @unroll for n = 1 : N ÷ 8
+                @unroll for k = 1 : K
+                    y_layout, x_layout = $convert_index_func((y + (k - 1), x + 8 * (n - 1)))
+                    @inbounds frag = setindex(frag, CT(workspace[y_layout, x_layout]), k + (K) * (n - 1))
+                end
+            end
+
+            return NTuple{K * N ÷ 8, CT}(frag)
+        end
+
+        @inline function load_c(::Type{FPUOp{M, N, K, DT, CT}}, ::Type{$layout_type{DT}}, workspace, tile::Tile) where {M, N, K, DT, CT}
+            laneId = (threadIdx().x - 1) % 32 + 1
+
+            op_y = (laneId - 1) ÷ 8 + 1
+            op_x = (laneId - 1) % 8 + 1
+
+            y, x = (tile.base.M + tile.offset.M + op_y, tile.base.N + tile.offset.N + op_x)
+
+            frag = LocalArray{Tuple{M ÷ 4, N ÷ 8}, DT}(undef)
+            @unroll for m = 1 : M ÷ 4
+                @unroll for n = 1 : N ÷ 8 
+                    @inbounds frag = setindex(frag, DT(workspace[y + 4 * (m - 1), x + 8 * (n - 1)]), m, n)
+                end
+            end
+
+            return NTuple{M * N ÷ 32, DT}(frag)
+        end
+
+        @inline function store_d(::Type{FPUOp{M, N, K, DT, CT}}, ::Type{$layout_type{DT}}, workspace, frag, tile::Tile) where {M, N, K, DT, CT}
+            laneId = (threadIdx().x - 1) % 32 + 1
+
+            op_y = (laneId - 1) ÷ 8 + 1
+            op_x = (laneId - 1) % 8 + 1
+
+            y, x = (tile.base.M + tile.offset.M + op_y, tile.base.N + tile.offset.N + op_x)
+
+            frag = LocalArray{Tuple{M ÷ 4, N ÷ 8}, DT}(frag)
+            @unroll for m = 1 : M ÷ 4
+                @unroll for n = 1 : N ÷ 8 
+                    @inbounds workspace[y + 4 * (m - 1), x + 8 * (n - 1)] = frag[m, n]
+                end
+            end
+        end
+    end
+end
+
+function mma(::Type{FPUOp{M, N, K, DT, CT}}, a_frag, b_frag, c_frag) where {M, N, K, DT, CT}
+    @unroll for k = 1 : K
+        @unroll for m = 1 : M ÷ 4
+            @unroll for n = 1 : N ÷ 8 
+                if (DT == CT)
+                    # Use fma
+                    @inbounds c_frag = setindex(
+                        c_frag,
+                        DT(fma(a_frag[m + (M ÷ 4) * (k - 1)], b_frag[k + K * (n - 1)], c_frag[m + (M ÷ 4) * (n - 1)])),
```

Comment:
```julia
    a_frag = LocalArray{Tuple{M ÷ 4, K}, CT}(a_frag)

    @unroll for k = 1 : K
        @unroll for m = 1 : M ÷ 4
            @unroll for n = 1 : N ÷ 8 
                @inbounds c_frag = setindex(
                    c_frag,
                    fma(
                        a_frag[m, k], 
                        b_frag[k + K * (n - 1)], 
                        c_frag[m + (M ÷ 4) * (n - 1)]
                    ),
                    m + (M ÷ 4) * (n - 1)
                )
            end
        end
    end

    return NTuple{M * N ÷ 32}(c_frag)
```
This, for example, requires 198 registers instead of 168 according to Nsight Compute, increasing the execution time by approx. 1000x.

Reply:
Can you diff the generated code? I had a quick look, and ND indexing of LocalArray seems to generate the same IR. Maybe this is just a missing `@inline`?

Reply:
`@inline` seems to have fixed it. Thank you.

Reply:
Great!